lib/rubocop/cop/rails/top_level_hash_with_indifferent_access.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Identifies top-level `HashWithIndifferentAccess`.
+      # This has been soft-deprecated since Rails 5.1.
+      #
+      # @example
+      #   # bad
+      #   HashWithIndifferentAccess.new(foo: 'bar')
+      #
+      #   # good
+      #   ActiveSupport::HashWithIndifferentAccess.new(foo: 'bar')
+      #
+      class TopLevelHashWithIndifferentAccess < Base
+        extend AutoCorrector
+        extend TargetRailsVersion
+
+        minimum_target_rails_version 5.1
+
+        MSG = 'Avoid top-level `HashWithIndifferentAccess`.'
+
+        # @!method top_level_hash_with_indifferent_access?(node)
+        #   @param [RuboCop::AST::ConstNode] node
+        #   @return [Boolean]
+        def_node_matcher :top_level_hash_with_indifferent_access?, <<~PATTERN
+          (const {nil? cbase} :HashWithIndifferentAccess)
+        PATTERN
+
+        # @param [RuboCop::AST::ConstNode] node
+        def on_const(node)
+          return unless top_level_hash_with_indifferent_access?(node)
+
+          add_offense(node) do |corrector|
+            autocorrect(corrector, node)
+          end
+        end
+
+        private
+
+        def autocorrect(corrector, node)
+          corrector.insert_before(
+            node.location.name,
+            'ActiveSupport::'
+          )
```

Comment:
Nits:

```suggestion
          corrector.insert_before(node.location.name, 'ActiveSupport::')
```

Reply:
Thanks.
BTW, can't we detect this kind of coding style preferences by RuboCop?

Reply:
I'm not looking for it :-)

Reply:
Hmm. I think there is a feature in Layout/LineLength or something Cop that autocorrects to wrap if the code is too long. If so, it would be nice to have the opposite one. If I come up with any good idea, I'll send a pull request to rubocop gem.

Reply:
@dvandersluis Great! I'll take it later, thank you!